Intermittent issue in create_raw_transaction                                        assert_equal(final_psbt["complete"], True) fails

https://cirrus-ci.com/task/4904241892425728?command=ci#L5098


```
 test  2021-01-20T23:39:58.801000Z TestFramework (ERROR): Assertion failed 
                                   Traceback (most recent call last):
                                     File "/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-i686-pc-linux-gnu/test/functional/test_framework/test_framework.py", line 126, in main
                                       self.run_test()
                                     File "/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-i686-pc-linux-gnu/test/functional/feature_nulldummy.py", line 94, in run_test
                                       test4tx = create_transaction(self.nodes[0], test2tx.hash, self.address, amount=46)
                                     File "/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-i686-pc-linux-gnu/test/functional/test_framework/blocktools.py", line 164, in create_transaction
                                       raw_tx = create_raw_transaction(node, txid, to_address, amount=amount)
                                     File "/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-i686-pc-linux-gnu/test/functional/test_framework/blocktools.py", line 180, in create_raw_transaction
                                       assert_equal(final_psbt["complete"], True)
                                     File "/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-i686-pc-linux-gnu/test/functional/test_framework/util.py", line 50, in assert_equal
                                       raise AssertionError("not(%s)" % " == ".join(str(arg) for arg in (thing1, thing2) + args))
                                   AssertionError: not(False == True)

The same error in the recent "Linux32" build on https://bitcoinbuilds.org/?job=a67a5e50-f422-436f-b173-54a4765094db:
```
...
2021-05-29T13:54:49.294000Z TestFramework (INFO): Initializing test directory /tmp/test_runner_₿_🏃_20210529_093535/feature_nulldummy_72
2021-05-29T13:54:51.887000Z TestFramework (INFO): Test 1: NULLDUMMY compliant base transactions should be accepted to mempool and mined before activation [103]
2021-05-29T13:54:51.908000Z TestFramework (INFO): Test 2: Non-NULLDUMMY base multisig transaction should not be accepted to mempool before activation
2021-05-29T13:54:51.914000Z TestFramework (INFO): Test 3: Non-NULLDUMMY base transactions should be accepted in a block before activation [104]
2021-05-29T13:54:51.917000Z TestFramework (INFO): Test 4: Non-NULLDUMMY base multisig transaction is invalid after activation
2021-05-29T13:54:51.920000Z TestFramework (ERROR): Assertion failed
Traceback (most recent call last):
  File "/home/ubuntu/src/test/functional/test_framework/test_framework.py", line 128, in main
    self.run_test()
  File "/home/ubuntu/src/test/functional/feature_nulldummy.py", line 95, in run_test
    test4tx = create_transaction(self.nodes[0], test2tx.hash, self.address, amount=46)
  File "/home/ubuntu/src/test/functional/test_framework/blocktools.py", line 165, in create_transaction
    raw_tx = create_raw_transaction(node, txid, to_address, amount=amount)
  File "/home/ubuntu/src/test/functional/test_framework/blocktools.py", line 181, in create_raw_transaction
    assert_equal(final_psbt["complete"], True)
  File "/home/ubuntu/src/test/functional/test_framework/util.py", line 50, in assert_equal
    raise AssertionError("not(%s)" % " == ".join(str(arg) for arg in (thing1, thing2) + args))
AssertionError: not(False == True)
2021-05-29T13:54:51.971000Z TestFramework (INFO): Stopping nodes
...
```

--- a/src/wallet/rpcwallet.cpp
+++ b/src/wallet/rpcwallet.cpp
@@ -4317,6 +4317,11 @@ static RPCHelpMan walletprocesspsbt()
     std::shared_ptr<CWallet> const pwallet = GetWalletForJSONRPCRequest(request);
     if (!pwallet) return NullUniValue;
 
+    const CWallet& wallet{*pwallet};
+    // Make sure the results are valid at least up to the most recent block
+    // the user could have gotten from another RPC command prior to now
+    wallet.BlockUntilSyncedToCurrentChain();
+
     RPCTypeCheck(request.params, {UniValue::VSTR, UniValue::VBOOL, UniValue::VSTR});
 
     // Unserialize the transaction
@@ -4333,7 +4338,7 @@ static RPCHelpMan walletprocesspsbt()
     bool sign = request.params[1].isNull() ? true : request.params[1].get_bool();
     bool bip32derivs = request.params[3].isNull() ? true : request.params[3].get_bool();
     bool complete = true;
-    const TransactionError err = pwallet->FillPSBT(psbtx, complete, nHashType, sign, bip32derivs);
+    const TransactionError err{wallet.FillPSBT(psbtx, complete, nHashType, sign, bip32derivs)};
     if (err != TransactionError::OK) {
         throw JSONRPCTransactionError(err);
     }
@@ -4431,6 +4436,11 @@ static RPCHelpMan walletcreatefundedpsbt()
     std::shared_ptr<CWallet> const pwallet = GetWalletForJSONRPCRequest(request);
     if (!pwallet) return NullUniValue;
 
+    CWallet& wallet{*pwallet};
+    // Make sure the results are valid at least up to the most recent block
+    // the user could have gotten from another RPC command prior to now
+    wallet.BlockUntilSyncedToCurrentChain();
+
     RPCTypeCheck(request.params, {
         UniValue::VARR,
         UniValueType(), // ARR or OBJ, checked later
@@ -4442,7 +4452,7 @@ static RPCHelpMan walletcreatefundedpsbt()
 
     CAmount fee;
     int change_position;
-    bool rbf = pwallet->m_signal_rbf;
+    bool rbf{wallet.m_signal_rbf};
     const UniValue &replaceable_arg = request.params[3]["replaceable"];
     if (!replaceable_arg.isNull()) {
         RPCTypeCheckArgument(replaceable_arg, UniValue::VBOOL);
@@ -4453,7 +4463,7 @@ static RPCHelpMan walletcreatefundedpsbt()
     // Automatically select coins, unless at least one is manually selected. Can
     // be overridden by options.add_inputs.
     coin_control.m_add_inputs = rawTx.vin.size() == 0;
-    FundTransaction(*pwallet, rawTx, fee, change_position, request.params[3], coin_control, /* override_min_fee */ true);
+    FundTransaction(wallet, rawTx, fee, change_position, request.params[3], coin_control, /* override_min_fee */ true);
 
     // Make a blank psbt
     PartiallySignedTransaction psbtx(rawTx);
@@ -4461,7 +4471,7 @@ static RPCHelpMan walletcreatefundedpsbt()
     // Fill transaction with out data but don't sign
     bool bip32derivs = request.params[4].isNull() ? true : request.params[4].get_bool();
     bool complete = true;
-    const TransactionError err = pwallet->FillPSBT(psbtx, complete, 1, false, bip32derivs);
+    const TransactionError err{wallet.FillPSBT(psbtx, complete, 1, false, bip32derivs)};
     if (err != TransactionError::OK) {
         throw JSONRPCTransactionError(err);
     }

--- a/test/functional/test_framework/blocktools.py
+++ b/test/functional/test_framework/blocktools.py
@@ -181,11 +181,6 @@ def create_raw_transaction(node, txid, to_address, *, amount):
             signed_psbt = wrpc.walletprocesspsbt(psbt)
             psbt = signed_psbt['psbt']
     final_psbt = node.finalizepsbt(psbt)
-    if not final_psbt["complete"]:
-        node.log.info(f'final_psbt={final_psbt}')
-        for w in node.listwallets():
-            wrpc = node.get_wallet_rpc(w)
-            node.log.info(f'listunspent={wrpc.listunspent()}')
     assert_equal(final_psbt["complete"], True)
     return final_psbt['hex']
 

